src/components/pages/contact/contact-form.tsx

```diff
@@ -1,148 +1,165 @@
 import React, { useState } from 'react';
+import { TextInput } from '../../forms/text-field/text-input';
+import { TextArea } from '../../forms/text-field/text-area';
+import { useTranslation } from 'gatsby-plugin-react-i18next';
 import { useForm } from 'react-hook-form';
-import {
-  ButtonText,
-  ErrorMessageSend,
-  InputField,
-  SendButton,
-  SentButton,
-  Sup,
-} from '../../legacy/form/controls';
-import { Grid, GridItem } from '../../legacy/grid/grid';
-import { Link } from '../../legacy/links/links';
-import { CaptionText } from '../../legacy/typography';
-import { Trans, useTranslation } from 'gatsby-plugin-react-i18next';
+import { Button } from '../../ui/buttons/button';
+import { SIMPLE_EMAIL_PATTERN } from '../../forms/constants';
 import styled from 'styled-components';
-import { SIMPLE_EMAIL_PATTERN } from '../../legacy/form/constants';
-import { CheckmarkIcon } from '../../legacy/icons/form-icons/checkmark';
-import { RightArrowIcon } from '../../legacy/icons/form-icons/right-arrow';
+import { TextStyles } from '../../typography';
+import { theme } from '../../layout/theme';
+import { Link } from '../../legacy/links/links';
+import { up } from '../../support/breakpoint';
 import { HoneypotField } from './honeypot';
 
-type RequestStatus = 'pending' | 'submitting' | 'success' | 'error';
 const API_ENDPOINT = '/api/contact-form';
 
-const StyledCaptionText = styled(CaptionText)`
-  color: inherit;
+const Grid = styled.div`
+  margin-bottom: 24px;
+  gap: 24px;
+  display: grid;
+  grid-template-columns: repeat(12, 1fr);
+`;
+
+const Caption = styled.p`
+  ${TextStyles.textS}
+  margin: 16px 0 48px;
+`;
+
+const ErrorMessage = styled.p`
+  ${TextStyles.textXS}
+  font-weight: 700;
+  display: inline-block;
+
+  color: ${theme.palette.text.errorMessage};
+`;
+
+const StyledButton = styled(Button)`
+  margin-right: 16px;
+`;
+
+const GridItem = styled.div`
+  grid-column-start: span 12;
+
+  ${up('md')} {
+    grid-column-start: span 6;
+  }
 `;
+
+const FirstName = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextInput
+      name={'name'}
+      label={t('contact.name')}
+      control={control}
+      rules={{ required: t<string>('contact.error.name') }}
+    />
+  );
+};
+
+const Email = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextInput
+      name={'email'}
+      label={t('contact.email')}
+      control={control}
+      rules={{
+        required: t<string>('contact.error.email'),
+        pattern: {
+          value: SIMPLE_EMAIL_PATTERN,
+          message: t<string>('contact.error.email-undefined'),
+        },
+      }}
+    />
+  );
+};
+
+const MessageArea = ({ control }) => {
+  const { t } = useTranslation();
+  return (
+    <TextArea
+      name={'message'}
+      label={t('contact.message')}
+      control={control}
+      rules={{ required: t<string>('contact.error.message') }}
+    />
+  );
+};
+
```

Comment:
I think you can collect those fields in a separate file `form-fields.tsx` to clean up the contact form file itself from the clutter.

Reply:
I think that helped a lot for readability and focus thank